src/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs

```diff
@@ -160,6 +160,17 @@ public static byte[] ConvertAlpnProtocolListToByteArray(List<SslApplicationProto
                     done = Interop.OpenSsl.DoSslHandshake(((SafeDeleteSslContext)context).SslContext, inputBuffer.token, inputBuffer.offset, inputBuffer.size, out output, out outputSize);
                 }
 
+                // When the handshake is done, and the context is server, check if the alpnHandle was freed during ALPN.
+                // If it was freed, then that indiciates ALPN failed, send failure.
```

Comment:
It wasn't freed, rather the Target was set to null.  There's a difference.  If the Target is set to null, the GCHandle still holds a slot and needs to be freed subsequently.